screens/AddUserToChatScreen.js

```diff
@@ -67,48 +62,48 @@ class AddUserToChatScreen extends React.Component {
     };
 
     _submitForm = async () => {
-        // clear previous error message
+        const { email } = this.state;
+        const { fetchFindUserByEmail } = this.props;
+
+        await fetchFindUserByEmail(email);
+
         this.setState({
-            errorMessage: ''
+            email: ''
         });
-        
-        const { email } = this.state;
-        const headers = await addAuthHeader();
-        return fetch(`${baseUrl}user/${email}`, {
-            headers
-        })
-            .then(handleResponse)
-            .then((data) => {
-                // if successful, clear email and update foundUser
-                this.setState({
-                    foundUser: data,
-                    email: ''
-                });
-            })
-            .catch(({ message }) => {
-                // if error, update error message
-                this.setState({
-                    errorMessage: message
-                });
-            });
     };
 
     componentDidUpdate(prevProps) {
+        const {
+            addedUser,
+            navigation,
+            error
+        } = this.props;
+
         // if adding user successful, display toast and navigate back to chat
-        if (prevProps.addedUser !== this.props.addedUser) {
+        if (addedUser && prevProps.addedUser !== addedUser) {
             Toast.show({
-                text: `${this.props.addedUser.displayName} added to chat`,
+                text: `${addedUser.displayName} added to chat`,
                 buttonText: "Okay",
                 type: "success",
                 duration: 2000,
-                onClose: () => this.props.navigation.navigate('Chat')
+                onClose: () => navigation.navigate('Chat')
+            });
+        }
+
+        // if error, display toast (and stay on same screen)
+        if (error && prevProps.error !== error) {
+            Toast.show({
+                text: `${error.message}; try again.`,
+                buttonText: "Okay",
+                type: "danger",
+                duration: 2000
             });
```

Comment:
There isn't a way to put the Toast in a component as far as I know unless I'm missing something? This is the basic pattern I followed to display a Toast with the error's message.

Reply:
hm i think you can, i just looked into it but went down a rabbit hole that i don't think is good for the last few days, so looks great! lol

Reply:
Hm yeah, I wasn't sure. It didn't seem like it and I def went down a rabbit hole trying to find a way, hahaha.